### ISSUE TITLE
'compute_rgbd_odometry' returns different transformation matrices on different platforms

### ISSUE THREAD

**Issue by botach** [intel-isl/Open3D#2522]
**EDITED  - After more debugging I edited the bug description**

**Describe the bug**
The function 'compute_rgbd_odometry' (under open3d.pipelines.odometry) returns two different transformation matrices on different platforms.
On the first platform, a MacBook Pro 16, the transformation matrix is precise and everything works fine, and the same result was reproduced on an Ubuntu 18.04 machine. On the other hand, on an Ubuntu 20.04 machine a completely different transformation matrix is returned which is nowhere near the correct transformation.

**Environments**

**Macbook Pro 16 Environment - works ok**
- Operating system: OS X Catalina (10.15.7), 16GB RAM, 1TB SSD, Intel Core i9 8-core CPU
- Python version: 3.8.5
- Open3D version: 0.11.0
- Is this remote workstation?: no
- How did you install Open3D?: pip (inside a conda environment)

**Linux PC 1 - works ok**
- Operating system: Ubuntu 18.04
- Python version: 3.8.5
- Open3D version: 0.11.0
- Is this a remote workstation?: no
- How did you install Open3D?: pip (inside a conda environment)

**Linux PC 2 - bad transformation is returned**
- Operating system: Ubuntu 20.04, 64GB RAM, 512GB SSD, Intel Core i9-10920X 3.5GHz 24-core CPU
- Python version: 3.8.5
- Open3D version: 0.11.0
- Is this a remote workstation?: no
- How did you install Open3D?: pip (inside a conda environment)

----------------------------------------------------
I should also mention that in an attempt to solve the problem I tried building Open3d 0.11.0 from source (on Linux PC 2 mentioned above) and ran my program again but the result was still the same.
This is the build config output:

```
cmake -DCMAKE_INSTALL_PREFIX=../installation ..
-- THIRD_PARTY_DOWNLOAD_DIR is not specified, will download directly.
-- No CMAKE_BUILD_TYPE specified, default to RelWithDebInfo
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Open3D 0.11.0+30329b2d
-- Using python from PATH: /home/user/miniconda3/envs/dataset/bin/python
-- Performing Test HAS_GLIBCXX_USE_CXX11_ABI
-- Performing Test HAS_GLIBCXX_USE_CXX11_ABI - Success
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Building library Assimp from source
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Building with OpenMP
-- Building library 3rdparty_eigen3 from source
-- Building library 3rdparty_flann from source
-- Building library 3rdparty_nanoflann from source
-- Building library 3rdparty_glew from source
-- Building library 3rdparty_glfw3 from source
-- Found Vulkan: /usr/lib/x86_64-linux-gnu/libvulkan.so  
-- Using X11 for window creation
-- Found X11: /usr/include   
-- Building third-party library JPEG from source
-- Looking for a ASM_NASM compiler
-- Looking for a ASM_NASM compiler - NOTFOUND
-- NASM assembler not found - libjpeg-turbo performance may suffer
-- libturbojpeg: WITH_CRT_DLL=OFF
-- Building library 3rdparty_jsoncpp from source
-- Building library 3rdparty_lzf from source
-- Building library 3rdparty_tritriintersect from source
-- Building third-party library zlib from source
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Building third-party library libpng from source
-- Building library 3rdparty_rply from source
-- Building library 3rdparty_tinyfiledialogs from source
-- Building library 3rdparty_tinygltf from source
-- Building library 3rdparty_tinyobjloader from source
-- Building library 3rdparty_qhull_r from source
-- Building library 3rdparty_qhullcpp from source
-- Building library 3rdparty_fmt from source
-- Found PythonInterp: /home/user/miniconda3/envs/dataset/bin/python (found version "3.8.5") 
-- Found PythonLibs: /home/user/miniconda3/envs/dataset/lib/libpython3.8.so
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- pybind11 v2.5.0
-- Building library 3rdparty_poisson from source
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Building library 3rdparty_imgui from source
-- Using prebuilt third-party library Filament
-- GLIBC version 2.31-0 found: Downloading Open3D Filament binary.
-- Filament is located at /home/user/Desktop/DCD/Open3D/build/filament/src/ext_filament
-- CLANG_LIBDIR: /usr/lib/x86_64-linux-gnu
-- CPP_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++.so
-- CPPABI_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++abi.so
-- Using remote url(s): https://anaconda.org/intel/mkl-static/2020.1/download/linux-64/mkl-static-2020.1-intel_217.tar.bz2
-- Using remote url(s): https://github.com/intel-isl/Open3D/releases/download/v0.10.0/linux-merged-mkl-static-2020.1-intel_217.zip
-- Using MKL to support BLAS and LAPACK functionalities.
-- STATIC_MKL_INCLUDE_DIR: /home/user/Desktop/DCD/Open3D/build/mkl_install/include/
-- STATIC_MKL_LIB_DIR: /home/user/Desktop/DCD/Open3D/build/mkl_install/lib
-- STATIC_MKL_LIBRARIES: mkl_merged;tbb_static
-- BUILD_JUPYTER_EXTENSION is set to OFF
-- Using Python version: 3.8
-- Using Python version major: 3
-- PYPI_PACKAGE_NAME: open3d
-- ========================================================================
-- Open3D 0.11.0 Configuration Summary
-- ========================================================================
-- Enabled Features:
--   OpenMP .................................. ON
--   Headless Rendering ...................... OFF
--   Azure Kinect Support .................... OFF
--   CUDA Support ............................ OFF
--   Build GUI ............................... ON
--   Build Shared Library .................... OFF
--   Build Unit Tests ........................ OFF
--   Build Examples .......................... 
--   Build Python Module ..................... ON
--   - with Jupyter Notebook Support ......... OFF
--   Build Tensorflow Ops .................... OFF
--   Build Pytorch Ops ....................... OFF
--   Build Benchmarks ........................ OFF
--   Bundle Open3D-ML ........................ OFF
--   Build RPC interface ..................... OFF
--   Force GLIBCXX_USE_CXX11_ABI= ............ 0
-- ========================================================================
-- Third-Party Dependencies:
--   EIGEN3 .................................. yes (build from source)
--   FILAMENT ................................ yes (build from source)
--   FLANN ................................... yes (build from source)
--   FMT ..................................... yes (build from source)
--   GLEW .................................... yes (build from source)
--   GLFW .................................... yes (build from source)
--   GOOGLETEST .............................. no
--   IMGUI ................................... yes (build from source)
--   JPEG .................................... yes (build from source)
--   JSONCPP ................................. yes (build from source)
--   LIBLZF .................................. yes (build from source)
--   OPENGL .................................. yes (build from source)
--   PNG ..................................... yes (build from source)
--   PYBIND11 ................................ yes (build from source)
--   QHULL ................................... yes (build from source)
--   LIBREALSENSE ............................ no
--   TINYFILEDIALOGS ......................... yes (build from source)
--   TINYGLTF ................................ yes (build from source)
--   TINYOBJLOADER ........................... yes (build from source)
-- ========================================================================
-- Configuring done
-- Generating done
```

Is it possible that there is a problem with one of the math libraries? (e.g. Eigen)

Thanks in advance :) 
I would like to thank all the Open3d developers for all your hard work, it is much appreciated 👍

**Comment by ssheorey** [comment]
Hi @botach thanks for reporting this. Can you share the stripped down data that gives different results?

**Comment by theNded** [comment]
Discussed in #2107 and fixed in #2567.
